packages/samples/headless-react/src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,373 @@
+  NumericFacet as HeadlessNumericFacet,
+  buildNumericFacet,
+  buildNumericRange,
+  buildDateSortCriterion,
+  buildFieldSortCriterion,
+  buildRelevanceSortCriterion,
+  Sort as HeadlessSort,
+  buildSort,
+  SortOrder,
+  SortCriterion,
+  ResultsPerPage as HeadlessResultsPerPage,
+  buildResultsPerPage,
+  Pager as HeadlessPager,
+  buildPager,
+  History as HeadlessHistory,
+  buildHistory,
+  RelevanceInspector as HeadlessRelevanceInspector,
+  buildRelevanceInspector,
+  AnalyticsActions,
+  SearchActions,
+  Unsubscribe,
+} from '@coveo/headless';
+import {bindSearchParametersToURI} from '../components/search-parameter-manager/search-parameter-manager';
+import {setContext} from '../components/context/context';
+
+const [KB, MB, GB] = [1e3, 1e6, 1e9];
+
+const criteria: [string, SortCriterion][] = [
+  ['Relevance', buildRelevanceSortCriterion()],
+  ['Date (Ascending)', buildDateSortCriterion(SortOrder.Ascending)],
+  ['Date (Descending)', buildDateSortCriterion(SortOrder.Descending)],
+  ['Size (Ascending)', buildFieldSortCriterion('size', SortOrder.Ascending)],
+  ['Size (Descending)', buildFieldSortCriterion('size', SortOrder.Descending)],
+];
+const initialCriterion = criteria[0][1];
+
+const resultsPerPageOptions = [10, 25, 50, 100];
+
+export class SearchPage extends Component {
+  private readonly engine: HeadlessEngine<typeof searchAppReducers>;
+  private readonly recommendationEngine: HeadlessEngine<
+    typeof recommendationAppReducers
+  >;
+
+  private readonly recommendationList: HeadlessRecommendationList;
+  private readonly tabs: {
+    all: HeadlessTab;
+    messages: HeadlessTab;
+    confluence: HeadlessTab;
+  };
+  private readonly searchBox: HeadlessSearchBox;
+  private readonly didYouMean: HeadlessDidYouMean;
+  private readonly searchStatus: HeadlessSearchStatus;
+  private readonly queryError: HeadlessQueryError;
+  private readonly querySummary: HeadlessQuerySummary;
+  private readonly resultList: HeadlessResultList;
+  private readonly facetManager: HeadlessFacetManager;
+  private readonly geographyFacet: HeadlessCategoryFacet;
+  private readonly objectTypeFacet: HeadlessFacet;
+  private readonly fileSizeAutomaticNumericFacet: HeadlessNumericFacet;
+  private readonly fileSizeManualNumericFacet: HeadlessNumericFacet;
+  private readonly sort: HeadlessSort;
+  private readonly resultsPerPage: HeadlessResultsPerPage;
+  private readonly pager: HeadlessPager;
+  private readonly history: HeadlessHistory;
+  private readonly relevanceInspector: HeadlessRelevanceInspector;
+
+  private stopUpdatingSearchParameters?: Unsubscribe;
+
+  constructor(props: {}) {
+    super(props);
+
+    this.engine = new HeadlessEngine({
+      configuration: HeadlessEngine.getSampleConfiguration(),
+      reducers: searchAppReducers,
+    });
+
+    this.recommendationEngine = new HeadlessEngine({
+      configuration: HeadlessEngine.getSampleConfiguration(),
+      reducers: recommendationAppReducers,
+    });
```

Comment:
Engines used to be initialized statically in `engine.ts`, now they are initialized in the search interface.